### PR TITLE
Jts postgis dependencies update

### PR DESCRIPTION
**Pull request by akatie**
update POM conflicting dependencies related to PostGres/PostGIS, JTS vividsolutions to locationtech, misc changes for ES 5.6.x, log4j update, leaving slf4j for next round

### CODE REVIEW

**Review by akatie — COMMENTED** [review]
JarHell is a hack; not sure how to address this otherwise

**Comment by akatie** [comment]
the updated JTS no longer needs WKT translation, so the test `dbutil` function for cast to geometry 'extractGeometry()' is now thinner, and may no longer be necessary.

File: src/test/java/de/komoot/photon/nominatim/testdb/H2DataAdapter.java [hunk 1]
```diff
@@ -1,8 +1,12 @@
...
 
-import com.vividsolutions.jts.geom.Geometry;
-import com.vividsolutions.jts.io.ParseException;
-import com.vividsolutions.jts.io.WKTReader;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.geom.GeometryFactory;
+import org.locationtech.jts.geom.Point;
+import org.locationtech.jts.geom.Polygon;
+import org.locationtech.jts.io.ParseException;
+import org.locationtech.jts.io.WKTReader;
```
Comment (lonvia) [inline src/test/java/de/komoot/photon/nominatim/testdb/H2DataAdapter.java:8]: Unnecessary new includes here.

File: src/test/java/org/elasticsearch/bootstrap/JarHellNo.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package org.elasticsearch.bootstrap;
+
+import java.net.URL;
+import java.util.Set;
+import com.google.common.collect.ImmutableSet;
+
+public class JarHellNo {
```
Comment (lonvia) [inline src/test/java/org/elasticsearch/bootstrap/JarHellNo.java:7]: Please remove this class completely.

File: pom.xml [hunk 3]
```diff
@@ -49,7 +48,7 @@
         <dependency>
             <groupId>org.postgresql</groupId>
             <artifactId>postgresql</artifactId>
-            <version>42.2.12</version>
+            <version>42.2.11</version>
```
Comment (lonvia) [inline pom.xml:51]: Why this version downgrade?

**Pull request closed without merging**